charts/gitness/templates/deployment.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "gitness.fullname" . }}
+  labels:
+    {{- include "gitness.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "gitness.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "gitness.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "gitness.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
```

Comment:
We will probably need dind-daemon to run for us to support running CI too. I tested it like this 

```        - name: dind-daemon
          image: docker:24.0.3-dind
          command: ["dockerd", "--host", "tcp://127.0.0.1:2375"]
          resources:
            requests:
              cpu: 20m
              memory: 512Mi
          lifecycle:
            postStart:
              exec:
                command:
                  - "/bin/sh"
                  - "-c"
                  - "echo '127.0.0.1 host.docker.internal' >> /etc/hosts"
          securityContext:
            privileged: true
          volumeMounts:
            - name: docker-data
              mountPath: /certs
```

also added 
```
          env:
            - name: DOCKER_HOST
              value: tcp://localhost:2375
```

in gitness container

Reply:
@abhinav-harness Do you want to add this as 2 containers or as a second pod that can then be accessed from gitness?

Reply:
@abhinav-harness I added it. The container can be deactivated via the Values files and the image can be exchanged via it.